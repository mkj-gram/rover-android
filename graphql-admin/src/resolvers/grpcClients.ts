import * as Auth from '@rover/auth-client'
export const authClient = Auth.v1.Client()
