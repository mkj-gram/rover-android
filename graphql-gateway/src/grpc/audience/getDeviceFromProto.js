import RoverApis from '@rover/apis'

const getVersionFromProto = p => !!p ? [p.getMajor(), p.getMinor(), p.getRevision()] : []

const getPlatformFromProto = p => {
    switch (p) {
        case 0:
            return 'UNDEFINED'
        case 1:
            return 'MOBILE'
        case 2:
            return 'WEB'
    }
}

export default (devices, d) => {
    devices[d.getProfileId()] = []

    const props = {
        created_at: RoverApis.Helpers.timestampFromProto(d.getCreatedAt()),
        app_version: d.getAppVersion(),
        device_manufacturer: d.getDeviceManufacturer(),
        device_model: d.getDeviceModel(),
        os_name: d.getOsName(),
        locale_language: d.getLocaleLanguage(),
        locale_region: d.getLocaleRegion(),
        locale_script: d.getLocaleScript(),
        is_wifi_enabled: d.getIsWifiEnabled(),
        is_cellular_enabled: d.getIsCellularEnabled(),
        screen_width: d.getScreenWidth(),
        screen_height: d.getScreenHeight(),
        carrier_name: d.getCarrierName(),
        radio: d.getRadio(),
        time_zone: d.getTimeZone(),
        is_bluetooth_enabled: d.getIsBluetoothEnabled(),
        advertising_id: d.getAdvertisingId(),
        ip: d.getIp(),
        location_latitude: d.getLocationLatitude(),
        location_longitude: d.getLocationLongitude(),
        os_version: getVersionFromProto(d.getOsVersion()),
        sdk_version: getVersionFromProto(d.getFrameworksMap().get('io.rover.Rover')),
        device_id: d.getDeviceId(),
        is_test_device: d.getIsTestDevice(),
        platform: getPlatformFromProto(d.getPlatform()),
        profile_id: d.getProfileId(),
        push_token_is_active: d.getPushTokenIsActive(),
        region_monitoring_mode: d.getRegionMonitoringMode(),
        is_background_enabled: d.getIsBackgroundEnabled(),
        is_location_monitoring_enabled: d.getIsLocationMonitoringEnabled(),
        push_environment: d.getPushEnvironment(),
        push_token_key: d.getPushTokenKey(),
        push_token_created_at: RoverApis.Helpers.timestampFromProto(d.getPushTokenCreatedAt()),
        push_token_updated_at: RoverApis.Helpers.timestampFromProto(d.getPushTokenUpdatedAt())
    }

    devices[d.getProfileId()] = Object.keys(props).map(p => {
        return {
            attribute: p,
            selector: 'DEVICE',
            value: props[p]
        }
    })
}
