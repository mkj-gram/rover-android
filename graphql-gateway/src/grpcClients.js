import promisify from '@rover-common/grpc-promisify'

import Audience from '@rover/audience-client'
import Auth from '@rover/auth-client'

export const audienceClient = Audience.v1.Client()

export const authClient = Auth.v1.Client()