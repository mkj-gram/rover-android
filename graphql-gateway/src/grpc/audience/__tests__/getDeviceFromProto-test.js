import getDeviceFromProto, {
    getPlatformFromProto,
    getPushEnvironmentFromProto,
    getVersionFromProto
} from '../getDeviceFromProto'
import RoverApis from '@rover/apis'
import promisify from '@rover-common/grpc-promisify'
import { audienceClient, authClient } from '../../../grpcClients'
promisify(audienceClient)

describe('getDeviceFromProto', async () => {

    it('should match snapshot', async () => {
        const authContext = new RoverApis.auth.v1.Models.AuthContext()
        authContext.setAccountId(1)
        authContext.setPermissionScopesList(["admin"])
        const request = new RoverApis.audience.v1.Models.GetDeviceRequest()
        request.setDeviceId("adevice00")
        request.setAuthContext(authContext)
        const response = await audienceClient.getDevice(request)
        const device = getDeviceFromProto(response.getDevice())

        expect(device).toMatchSnapshot()
    })
})

describe('getPlatformFromProto', () => {
    it('should return empty string for platform 0', () => {
        const platform = getPlatformFromProto(0)
        expect(platform).toEqual('')
    }) 
    it('should return "Mobile" for platform 1', () => {
        const platform = getPlatformFromProto(1)
        expect(platform).toEqual('Mobile')
    }) 
    it('should return "Web" for platform 2', () => {
        const platform = getPlatformFromProto(2)
        expect(platform).toEqual('Web')
    }) 
})

describe('getPushEnvironmentFromProto', () => {
    it('should return a string with the first letter capitalized', () => {
        const pushEnvironment = getPushEnvironmentFromProto('development')
        expect(pushEnvironment).toEqual('Development')
    })
})

describe('getVersionFromProto', () => {
    it('should return an empty array for null or undefined proto version messages', () => {
        const version = getVersionFromProto(null)
        expect(version).toEqual([])
    }) 
    it('should return the major, minor, and revision values of a proto version object', () => {
        const protoVersion = new RoverApis.audience.v1.Models.Version()
        protoVersion.setMajor(1)
        protoVersion.setMinor(2)
        protoVersion.setRevision(3)
        const version = getVersionFromProto(protoVersion)
        expect(version).toEqual([1,2,3])
    })
})
