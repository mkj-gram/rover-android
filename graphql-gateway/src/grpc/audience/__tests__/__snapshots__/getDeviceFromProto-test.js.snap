// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDeviceFromProto should match snapshot 1`] = `
Array [
  Object {
    "attribute": "created_at",
    "selector": "DEVICE",
    "value": 2017-06-14T15:44:18.496Z,
  },
  Object {
    "attribute": "app_version",
    "selector": "DEVICE",
    "value": "1.0",
  },
  Object {
    "attribute": "device_manufacturer",
    "selector": "DEVICE",
    "value": "Apple",
  },
  Object {
    "attribute": "device_model",
    "selector": "DEVICE",
    "value": "iPhone7,2",
  },
  Object {
    "attribute": "os_name",
    "selector": "DEVICE",
    "value": "iOS",
  },
  Object {
    "attribute": "locale_language",
    "selector": "DEVICE",
    "value": "en",
  },
  Object {
    "attribute": "locale_region",
    "selector": "DEVICE",
    "value": "us",
  },
  Object {
    "attribute": "locale_script",
    "selector": "DEVICE",
    "value": "zz",
  },
  Object {
    "attribute": "is_wifi_enabled",
    "selector": "DEVICE",
    "value": true,
  },
  Object {
    "attribute": "is_cellular_enabled",
    "selector": "DEVICE",
    "value": true,
  },
  Object {
    "attribute": "screen_width",
    "selector": "DEVICE",
    "value": 1080,
  },
  Object {
    "attribute": "screen_height",
    "selector": "DEVICE",
    "value": 720,
  },
  Object {
    "attribute": "carrier_name",
    "selector": "DEVICE",
    "value": "rogers",
  },
  Object {
    "attribute": "radio",
    "selector": "DEVICE",
    "value": "LTE",
  },
  Object {
    "attribute": "time_zone",
    "selector": "DEVICE",
    "value": "America/Toronto",
  },
  Object {
    "attribute": "is_bluetooth_enabled",
    "selector": "DEVICE",
    "value": true,
  },
  Object {
    "attribute": "advertising_id",
    "selector": "DEVICE",
    "value": "",
  },
  Object {
    "attribute": "ip",
    "selector": "DEVICE",
    "value": "",
  },
  Object {
    "attribute": "location_latitude",
    "selector": "DEVICE",
    "value": 22.00101,
  },
  Object {
    "attribute": "location_longitude",
    "selector": "DEVICE",
    "value": -32.1231,
  },
  Object {
    "attribute": "os_version",
    "selector": "DEVICE",
    "value": Array [
      1,
      2,
      3,
    ],
  },
  Object {
    "attribute": "sdk_version",
    "selector": "DEVICE",
    "value": Array [],
  },
  Object {
    "attribute": "device_id",
    "selector": "DEVICE",
    "value": "adevice00",
  },
  Object {
    "attribute": "is_test_device",
    "selector": "DEVICE",
    "value": false,
  },
  Object {
    "attribute": "label",
    "selector": "DEVICE",
    "value": "",
  },
  Object {
    "attribute": "platform",
    "selector": "DEVICE",
    "value": "Web",
  },
  Object {
    "attribute": "profile_id",
    "selector": "DEVICE",
    "value": "00000000000000000000aaa2",
  },
  Object {
    "attribute": "push_token_is_active",
    "selector": "DEVICE",
    "value": true,
  },
  Object {
    "attribute": "region_monitoring_mode",
    "selector": "DEVICE",
    "value": 2,
  },
  Object {
    "attribute": "is_background_enabled",
    "selector": "DEVICE",
    "value": true,
  },
  Object {
    "attribute": "is_location_monitoring_enabled",
    "selector": "DEVICE",
    "value": true,
  },
  Object {
    "attribute": "push_environment",
    "selector": "DEVICE",
    "value": "development",
  },
  Object {
    "attribute": "push_token_key",
    "selector": "DEVICE",
    "value": "abc",
  },
  Object {
    "attribute": "push_token_created_at",
    "selector": "DEVICE",
    "value": 2017-06-14T15:44:18.496Z,
  },
  Object {
    "attribute": "push_token_updated_at",
    "selector": "DEVICE",
    "value": 2017-06-14T15:44:18.496Z,
  },
]
`;
